### PR TITLE
Remove telemetryevent.ts

### PR DESCRIPTION
**Pull request by hunterwerlla**
- Remove extraneous `TelemetryEvent`
- Record single events instead of arrays of events
- Cleanup some recording logic
## Testing
Confirmed read the cache properly/ignored invalid cache properly/posted properly
## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: src/shared/telemetry/defaultTelemetryService.ts [hunk 1]
```diff
@@ -295,25 +278,15 @@ export function filterTelemetryCacheEvents(input: any): TelemetryEvent[] {
...
-                return false
-            }
-
-            return true
-        })
-        .filter((item: TelemetryEvent) => {
             // skip it if data is not an array or empty
-            if (!Array.isArray(item.data) || item.data.length === 0) {
-                getLogger().warn(`Item in telemetry cache: ${item}\n has invalid data field: ${item.data}! skipping!`)
+            if (!Array.isArray(item) || item.length === 0) {
```
Comment (rli) [inline src/shared/telemetry/defaultTelemetryService.ts:282]: do we need a test to make sure we gracefully handle cached entries created before this change?

Comment (hunterwerlla) [reply src/shared/telemetry/defaultTelemetryService.ts:282]: we gracefully handle them by throwing them out, we have 4 generations of this cache so far and have never retained the old versions. The test will be updated to make sure it handles it properly.

**Review by justinmk3 — APPROVED** [review]

File: src/shared/telemetry/defaultTelemetryService.ts [hunk 2]
```diff
@@ -278,37 +278,24 @@ export function filterTelemetryCacheEvents(input: any): MetricDatum[] {
...
-            if (!Array.isArray(item) || item.length === 0) {
-                getLogger().warn(`Item in telemetry cache: ${item}\n has invalid data field: ${item}! skipping!`)
+            // Only accept objects that have the required telemetry data
+            if (
+                !item.hasOwnProperty('Value') ||
+                !item.hasOwnProperty('MetricName') ||
+                !item.hasOwnProperty('EpochTimestamp') ||
+                !item.hasOwnProperty('Unit')
+            ) {
+                getLogger().warn(`Item in telemetry cache: ${JSON.stringify(item)}\n has invalid data! skipping!`)
```
Comment (justinmk3) [inline src/shared/telemetry/defaultTelemetryService.ts:288]:
```suggestion
                getLogger().warn(`skipping invalid telemetry cache item: ${JSON.stringify(item)}\n`)
```

In general it is useful to put the variable part at the end of a log message. Especially in this case, where the variable part is multiline, having to scan the whole log to find the "skipping!" message at the end adds friction.

**Review by justinmk3 — APPROVED** [review]

**Review by justinmk3 — APPROVED** [review]

**Pull request merged**